### PR TITLE
Implementation of `eth_sendTransaction`

### PR DESCRIPTION
**Pull request by goran-ethernal**
# Description

This PR implements `eth_sendTransaction` json-rpc endpoint.

`eth_sendTransaction` creates new message call transaction or a contract creation, if the data field contains code, and signs it using the account specified in from.

**Parameters**

Object - The transaction object

Example:

```
params: [
  {
    from: "0xb60e8dd61c5d32be8058bb8eb970870f07233155",
    to: "0xd46e8dd67c5d32be8058bb8eb970870f07244567",
    gas: "0x76c0", // 30400
    gasPrice: "0x9184e72a000", // 10000000000000
    value: "0x9184e72a", // 2441406250
    input:
      "0xd46e8dd67c5d32be8d46e8dd67c5d32be8058bb8eb970870f072445675058bb8eb970870f072445675",
  },
]
```

**Returns**

DATA, 32 Bytes - the transaction hash, or the zero hash if the transaction is not yet available.

Example:

```
// Request
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_sendTransaction","params":[{see above}],"id":1}'
// Result
{
  "id":1,
  "jsonrpc": "2.0",
  "result": "0xe670ec64341771606e55d6b4ca35a1a6b75ee3d5145a99d05921026d1527331"
}
```

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### CODE REVIEW

File: jsonrpc/eth_endpoint.go [hunk 1]
```diff
@@ -360,10 +348,19 @@ func (e *Eth) SendRawTransaction(buf argBytes) (interface{}, error) {
 	return tx.Hash().String(), nil
 }
 
-// SendTransaction rejects eth_sendTransaction json-rpc call as we don't support wallet management
-func (e *Eth) SendTransaction(_ *txnArgs) (interface{}, error) {
-	return nil, fmt.Errorf("request calls to eth_sendTransaction method are not supported," +
-		" use eth_sendRawTransaction instead")
+// SendTransaction creates a transaction for the given argument, sign it and submit it to the tx pool
```
Comment (Stefan-Ethernal) [inline jsonrpc/eth_endpoint.go]: Minor typos:
```suggestion
// SendTransaction creates a transaction for the given argument, signs it, and submits it to the tx pool
```

File: e2e-polybft/e2e/jsonrpc_test.go [hunk 2]
```diff
@@ -427,6 +427,46 @@ func TestE2E_JsonRPC(t *testing.T) {
...
+
+		nonce, err := ethClient.GetNonce(preminedAcctTwo.Address(), jsonrpc.LatestBlockNumberOrHash)
+		require.NoError(t, err)
+
+		gasPrice, err := ethClient.GasPrice()
+		require.NoError(t, err)
+
+		target := preminedAcctOne.Address()
+
+		txn := &jsonrpc.CallMsg{
```
Comment (Stefan-Ethernal) [inline e2e-polybft/e2e/jsonrpc_test.go]: Why you didn't want to instantiate `types.Transaction` here? If I were you, I'd do it and remove the `SendTransactionCallMsg` from the `EthClient`. WDYT?

**Review by Stefan-Ethernal — APPROVED** [review]

File: e2e-polybft/e2e/jsonrpc_test.go (hunk 2 above)
Comment (goran-ethernal) [reply e2e-polybft/e2e/jsonrpc_test.go]: Done: https://github.com/Ethernal-Tech/blade/pull/294/commits/7e8bf9a3735807398a557d7bac1b5dbb37fdb5fe

File: jsonrpc/eth_endpoint.go (hunk 1 above)
Comment (goran-ethernal) [reply jsonrpc/eth_endpoint.go]: Fix: https://github.com/Ethernal-Tech/blade/pull/294/commits/7e8bf9a3735807398a557d7bac1b5dbb37fdb5fe

**Review by oliverbundalo — APPROVED** [review]

**Pull request merged**